### PR TITLE
KEYCLOAK-18466 Configure HTTP client timeouts for adapters

### PR DESCRIPTION
**Pull request by mabartos**
JIRA: [KEYCLOAK-18466](https://issues.redhat.com/browse/KEYCLOAK-18466)

### CODE REVIEW

**Review by andymunro — DISMISSED** [review]

File: securing_apps/topics/oidc/java/java-adapter-config.adoc [hunk 1]
```diff
@@ -150,6 +153,19 @@ connection-pool-size::
   This is _OPTIONAL_.
   The default value is `20`.
 
+socket-timeout-millis::
+  Timeout for socket waiting for data after establishing the connection in milliseconds.
+  Maximum time of inactivity between two data packets.
```
Comment (mhajas) [inline securing_apps/topics/oidc/java/java-adapter-config.adoc:158]:
```suggestion
  Maximum time of inactivity between two data packets. A timeout value of zero is interpreted as an infinite timeout.
```

File: securing_apps/topics/oidc/java/java-adapter-config.adoc [hunk 2]
```diff
@@ -150,6 +153,19 @@ connection-pool-size::
...
+  This is _OPTIONAL_.
+
+connection-timeout-millis::
+  Timeout for establishing the connection with the remote host in milliseconds.
```
Comment (mhajas) [inline securing_apps/topics/oidc/java/java-adapter-config.adoc:165]:
```suggestion
  Timeout for establishing the connection with the remote host in milliseconds. A value of zero means the timeout is not used. The default value is zero.
```

**Review by mhajas — COMMENTED** [review]
Thank you for the PR @mabartos. I added two suggestions with the addition of more details to properties. Also, this will probably require other changes based on the last comment in the `keycloak/keycloak` PR

File: securing_apps/topics/oidc/java/java-adapter-config.adoc (hunk 1 above)
Comment (mabartos) [reply securing_apps/topics/oidc/java/java-adapter-config.adoc:158]: Done

File: securing_apps/topics/oidc/java/java-adapter-config.adoc (hunk 2 above)
Comment (mabartos) [reply securing_apps/topics/oidc/java/java-adapter-config.adoc:165]: It should be the same like `socket-timeout-millis`... There's used `org.apache.http.params.CoreConnectionPNames`, where the parameter has same behavior like the mentioned `socket-timeout-millis`. It should be: _A timeout value of zero is interpreted as an infinite timeout._

Comment (hmlnarik) [reply securing_apps/topics/oidc/java/java-adapter-config.adoc:165]: @mabartos Makes sense to keep this the same as in httpclient docs. 
The other point is valid, the default values should be documented similarly to the other options.

**Review by hmlnarik — APPROVED** [review]

**Pull request merged**